framework/src/engine/legacy/storage.ts

```diff
@@ -12,51 +12,78 @@
  * Removal or modification of this copyright notice is prohibited.
  */
 
-import { Database } from '@liskhq/lisk-db';
-
-export const DB_KEY_BLOCK_ID = Buffer.from([0]);
+import { intToBuffer } from '@liskhq/lisk-cryptography/dist-node/utils';
+import { Batch, Database } from '@liskhq/lisk-db';
+import { DB_KEY_BLOCK_HEIGHT, DB_KEY_BLOCK_ID } from './constants';
 
 export class Storage {
 	private readonly _db: Database;
 
 	public constructor(db: Database) {
 		this._db = db;
-		// eslint-disable-next-line no-console
-		console.log(this._db);
 	}
 
-	// eslint-disable-next-line @typescript-eslint/require-await
-	public async getBlockByID(_id: Buffer): Promise<Buffer> {
-		return Buffer.alloc(0);
+	public async getBlockByID(id: Buffer): Promise<Buffer> {
+		return this._db.get(Buffer.concat([DB_KEY_BLOCK_ID, id]));
 	}
 
-	// eslint-disable-next-line @typescript-eslint/require-await
-	public async getBlockByHeight(_height: number): Promise<Buffer> {
-		return Buffer.alloc(0);
+	public async getBlockByHeight(height: number): Promise<Buffer> {
+		return this.getBlockByID(
+			await this._db.get(Buffer.concat([DB_KEY_BLOCK_HEIGHT, intToBuffer(height, 4)])),
+		);
 	}
 
-	// eslint-disable-next-line @typescript-eslint/require-await
-	public async getBlocksByHeightBetween(_fromHeight: number, _toHeight: number): Promise<Buffer[]> {
-		return [];
+	public async getBlocksByHeightBetween(fromHeight: number, toHeight: number): Promise<Buffer[]> {
+		const blocks: Buffer[] = [];
+
+		await Promise.all(
+			(await this._getBlockIDsBetweenHeights(fromHeight, toHeight)).map(async id => {
+				blocks.push(await this.getBlockByID(id));
+			}),
+		);
```

Comment:
```suggestion
		const blockIDs = await this._getBlockIDsBetweenHeights(fromHeight, toHeight);
		const blocks = await Promise.all(blockIDs.map(id => this.getBlockByID(id)));
```
if you await inside, Promise.all wont work as expected

Reply:
Since am calling `getBlockByID` which returns a promise, It must be async right 🤔. 
`return Promise.all(blockIDs.map(async id => this.getBlockByID(id)));`